game/scripts/vscripts/Sections/Chapter6/SectionClosing.ts

```diff
@@ -104,22 +108,26 @@ class ClosingNpc {
     }
 
     interact() {
-        if (this.spawned && this.unit && unitIsValidAndAlive(this.unit)) {
-            // Play dialog
-            if (this.soundName) {
-                this.dialogToken = dg.playAudio(this.soundName, this.text, this.unit, undefined, () => this.dialogToken = undefined)
-            } else {
-                this.dialogToken = dg.playText(this.text, this.unit, 5, () => this.dialogToken = undefined)
+        if (canInteract) {
```

Comment:
Could you please move this if statement to the order filter? this is the wrong place to put it imo